.cbmc-batch/source/make_common_data_structures.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openssl/ec.h>
+#include <openssl/evp.h>
+
+#include <cipher_openssl.h>
+#include <ec_utils.h>
+#include <evp_utils.h>
+#include <make_common_data_structures.h>
+
+void ensure_md_context_has_allocated_members(struct aws_cryptosdk_md_context *ctx) {
+    ctx->alloc      = nondet_bool() ? NULL : can_fail_allocator();
+    ctx->evp_md_ctx = evp_md_ctx_nondet_alloc();
+}
+
+void ensure_sig_ctx_has_allocated_members(struct aws_cryptosdk_sig_ctx *ctx) {
+    ctx->alloc = nondet_bool() ? NULL : can_fail_allocator();
+    enum aws_cryptosdk_alg_id alg_id;
+    ctx->props   = aws_cryptosdk_alg_props(alg_id);
```

Comment:
Does this have any side-effects?

Reply:
Also, you could consider using a ```nondet_aws_cryptosdk_alg_id()```

Reply:
`aws_cryptosdk_alg_props` doesn't have side-effects, it just returns an `alg_properties` object